src/Microsoft.TestPlatform.Common/Utilities/ArgumentParser.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Utility methods for Parsing arguments via command line or run settings.
+    /// </summary>
+    public static class ArgumentParser
+    {
+        /// <summary>
+        /// Parses the parameters passed as name values pairs along with the logger argument.
+        /// </summary>
+        /// <param name="argument">Logger argument</param>
+        /// <param name="loggerIdentifier">Receives logger Uri or friendly name.</param>
+        /// <param name="parameters">Receives parse name value pairs.</param>
+        /// <returns>True is successful, false otherwise.</returns>
+        public static bool TryParseLoggerArgument(string argument, out string loggerIdentifier, out Dictionary<string, string> parameters)
+        {
+            loggerIdentifier = null;
+            parameters = null;
+
+            var parseSucceeded = true;
```

Comment:
kinda strange, you are staring with assumption that parseSucceeded = true, & setting it to false if it fails, what if you start it as false, then you can save last else statement